pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -413,18 +392,47 @@ func (o *OpenIDCProvider) getUserInfo(ctx *context.Context, config *v32.OIDCConf
 }
 
 func ConfigToOauthConfig(endpoint oauth2.Endpoint, config *v32.OIDCConfig) oauth2.Config {
+	var finalScopes []string
 	hasOIDCScope := strings.Contains(config.Scopes, oidc.ScopeOpenID)
 	// scopes must be space separated in string when passed into the api
 	configScopes := strings.Split(config.Scopes, " ")
 	if !hasOIDCScope {
 		configScopes = append(configScopes, oidc.ScopeOpenID)
 	}
-
+	for _, scope := range configScopes {
+		if scope != "" {
+			finalScopes = append(finalScopes, scope)
+		}
+	}
 	return oauth2.Config{
 		ClientID:     config.ClientID,
 		ClientSecret: config.ClientSecret,
 		Endpoint:     endpoint,
 		RedirectURL:  config.RancherURL,
-		Scopes:       configScopes,
+		Scopes:       finalScopes,
+	}
+}
+
+func (o *OpenIDCProvider) getGroupsFromClaimInfo(claimInfo claimInfo) []v3.Principal {
+	var groupPrincipals []v3.Principal
+
```

Comment:
Is there a possibility of nil claimInfo?

Reply:
The struct does not have "omitEmpty" on any of the fields, so at most it would be a bunch of empty values. I check the error when unmarshaling values into the struct so that should keep it from being nil.